src/ergw_gtp_c_socket.erl

```diff
@@ -143,6 +144,8 @@ init(#{name := Name, ip := IP, burst_size := BurstSize,
        send_port := SendPort} = SocketOpts) ->
     process_flag(trap_exit, true),
 
+    {ok, #{icmp_error_handling := Act}} = application:get_env(ergw, path_management),
```

Comment:
My previous comments might have been a bit confusing, they were meant to give alternatives not that all of them have to used at once.

So, the icmp_error renaming is fine, but:
a) either continue checking the  icmp_error_handling in the socket logic, if you choose that, then the config options has to go into the socket options
b) check the option only in the gtp_path logic, then the option can in stay path_management, no check need to be performed in the socket logic.

I would prefer variant b).

Reply:
Thanks for clarification, done.